source/common/upstream/health_checker_event_logger.cc

```diff
@@ -37,6 +37,13 @@ void HealthCheckEventLoggerImpl::logAddHealthy(
   });
 }
 
+void HealthCheckEventLoggerImpl::logSuccessfulHealthCheck(
+    envoy::data::core::v3::HealthCheckerType health_checker_type,
+    const HostDescriptionConstSharedPtr& host) {
+  createHealthCheckEvent(health_checker_type, *host,
+                         [](auto& event) { event.mutable_successful_health_check_event(); });
```

Comment:
This looks like a no-op, and what is this `mutable_successful_health_check_event` for?

Reply:
when calling ``mutable_x`` it sets the kind of the event type to this (similar as the ``mutable_degraded_healthy_host`` below). I intentionally created another kind of event type, to be able to distinguish between ``add_healthy`` and ``successful_health_check``.

Reply:
Thanks, could you add a test to cover this and verify the output similar [as this one](https://github.com/envoyproxy/envoy/blob/8318716d9aedfc6277cd605a41b606a86f3feb52/test/common/upstream/health_checker_impl_test.cc#L6468)?